### PR TITLE
Pass User Story 58 - Admin Visiting Merchant on User Show redirects to Merchant Show

### LINKED ISSUE #12: User Story 58, Admin is redirected from User profile to Merchant dashboard
```
As an admin user
If I visit a profile page for a user, but that user is a merchant
Then I am redirected to the appropriate merchant dashboard page.
eg, if I visit "/admin/users/7" but that user is a merchant
Then I am redirected to "/admin/merchants/7"
And I see their merchant dashboard page
```

DO THESE STORIES FIRST:
- User Story 56, Admin User Profile Page
- User Story 38, Admin can see a merchant's dashboard

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #16

child of #82

depends on #14

depends on #36

### PR DESCRIPTION
**Pull request by blake-enyart**
- [] Check this if the PR has been approved by the team to be a quick patch and the rest of this form will not be filled out

# Test Coverage Percentage (Model/Feature):
- rspec - 100%
- rspec spec/models - 100%

# What functionality does this accomplish?
* Redirects to merchant dashboard when admin visits user show page that belongs to merchant

# What did you struggle on to complete?
nada

# Implements/Fixes:
* description
closes #12 

# Testing Changes
- [X] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [X] My code has no unused/commented out code
- [X] I have reviewed my code
- [] I have commented my code, particularly in hard-to-understand areas
- [X] I have fully tested my code

# Please include an emoji of how you feel about this branch:
# 💯

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -2,6 +2,9 @@ class Admin::UsersController < Admin::BaseController
 
   def show
     @user = User.find(params[:id])
+    if @user.role == 'merchant'
```
Comment (milevy1) [inline app/controllers/admin/users_controller.rb]: We could use the enum helper:

`if @user.merchant?`

Comment (blake-enyart) [reply app/controllers/admin/users_controller.rb]: Just implemented this and we should be good!

**Comment by blake-enyart** [comment]
@milevy1 I refactored up here on github and it should be good to go now.
